### DISCUSSION TITLE
Model Refitting using Entire Dataset

### DISCUSSION THREAD

**Discussion by isachng93** [winedarksea/AutoTS#171] [General]
One query: After selecting the best model based on test set accuracy, would the selected model be refitted using the entire dataset, inclusive of the test data? 

--> In a similar way to what Modeltime packages does as described here: https://business-science.github.io/modeltime/reference/modeltime_refit.html

Thanks!

**Comment by winedarksea** [comment]
Yes, it uses the entire dataset for the final forecast. That is done in the `.predict` part. The `.fit` part only looks at the past in cross validation.

If you are looking at running AutoTS, say, weekly, I recommend you check out `production_example.py` in this repo. If you set `evolve=False` it will just run the same model (from template) each week, after initial training. With `evolve=True` there it will use the previous week as a base then try new additional models as well.

**Reply by winedarksea** [reply]
You can also use the `model_forecast` function with the best_model from the AutoTS class, it looks like the closest match to the model refit you showed there
